### ISSUE TITLE
Job Tasks

### ISSUE THREAD

**Issue by Imagine3D-Ben** [RinsibleElk/Deadfile#15]
What is the date on the job tasks for? Is this a due date? Also can this have an due/completed status please?

**Comment by Imagine3D-Ben** [comment]
OK so aside from some minor design points (e.g. colouring the tasks that have run past the date, completed tasks etc.) I think I love it. Just a few questions/comments.

I assume that the fact the tasks expand when they are selected is to ultimately add the notes field below them? If so then cool!

I love the navigation to Job/Client from the report that's awesome!

Not sure about the need to email it really but printing is a must. Generally this list is just for me but ultimately will give Dad the ability to add tasks when he is in the office. For when Dad is away we generally just speak on the phone or email about what needs to be done. If I need to email him the list (which is unlikely) I'd just print to PDF.

If possible can we filter by job and client too?

I assume you know the include inactive doesn't work yet.

**Comment by Imagine3D-Ben** [comment]
Oh also do you know the date filter only applies to completed jobs?

**Comment by Imagine3D-Ben** [comment]
Oh... include inactive does work sorry

**Comment by RinsibleElk** [comment]
The include inactive does work, however there is some complex logic with
the start and end date which might be making that a bit weird. Could be I
need to tweak the behaviour.

Filter by job/client is 2 minutes work. Consider it done, I'll probs add
two new text boxes for it.

The notes already show up if you expand.

Glad you like it, I don't like DataGrid for editing very much but was very
happy with how it came out for reporting.

**Comment by RinsibleElk** [comment]
Yeah that's the point. It occurred to me, why would you ever want to filter
out expired tasks. So I basically made it so the start date filter applies
only if include inactive is enabled and the job is dead, ie you just want
to cum over how much you got done this week.

**Comment by Imagine3D-Ben** [comment]
The way I saw the filter date thing working was

a) Filtering out active tasks too far in the future
b) Filtering out tasks that aren't due today
and 
c) Finding out what day I did something

What do you think about a priority tag? Just so if I have 20 tasks due today I can choose which ones I can leave until tomorrow if necessary.

**Comment by RinsibleElk** [comment]
Sure, say low medium high, affects ordering and maybe some indicator in the
template eg (mirroring outlook) a red exclamation mark or a blue downward
arrow. Sound OK?

On 14 Dec 2016 7:54 pm, "Imagine3D-Ben" <notifications@github.com> wrote:

The way I saw the filter date thing working was

a) Filtering out active tasks too far in the future
b) Filtering out tasks that aren't due today
and
c) Finding out what day I did something

What do you think about a priority tag? Just so if I have 20 tasks due
today I can choose which ones I can leave until tomorrow if necessary.

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/RinsibleElk/Deadfile/issues/15#issuecomment-267138246>,
or mute the thread

**Comment by Imagine3D-Ben** [comment]
Perfect!

**Comment by RinsibleElk** [comment]
For the filtering dates thing, the way I was presenting it is basically
like saying something that was due yday that is still active is very due
today. That was the point. I can do whatever behaviour you want, but my
point was that if I give you start and end dates, at a bare minimum if
start date is today or in the past then I had better include yesterday's
remaining in that?

On 14 Dec 2016 8:02 pm, "Oliver Samson" <oliver.samson@cantab.net> wrote:

Sure, say low medium high, affects ordering and maybe some indicator in the
template eg (mirroring outlook) a red exclamation mark or a blue downward
arrow. Sound OK?

On 14 Dec 2016 7:54 pm, "Imagine3D-Ben" <notifications@github.com> wrote:

The way I saw the filter date thing working was

a) Filtering out active tasks too far in the future
b) Filtering out tasks that aren't due today
and
c) Finding out what day I did something

What do you think about a priority tag? Just so if I have 20 tasks due
today I can choose which ones I can leave until tomorrow if necessary.

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/RinsibleElk/Deadfile/issues/15#issuecomment-267138246>,
or mute the thread

**Comment by Imagine3D-Ben** [comment]
OK that's not quite what I had imagined. I thought that if I was ever viewing an inactive task the point would be to find out the date whereas if I was filtering by date it was because I either didn't know what task was due to today or to ignore things that were too far ahead to worry about.

I do agree that dates in the past should be included today if they are still active though.
